### PR TITLE
Add content_owner.references to list References

### PR DESCRIPTION
**Pull request by stavro**
Add content_owner.references to list References

A new Reference class is introduced to list YouTube ContentID References.
See https://developers.google.com/youtube/partner/docs/v1/references

content_owner.references **must be given** a `where` clause to specify a comma-separated list of IDs, or an asset ID:

``` ruby
    content_owner.references.where id: 'z123,z234'
    content_owner.references.where asset_id: 'A123'
```

### CODE REVIEW

**Comment by stavro** [comment]
Only thing missing from references (which I can add in a different PR) is `excluded_intervals`

image

File: lib/yt/models/reference.rb [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      #   checking, computing_fingerprint, deleted, duplicate_on_hold, inactive, 
+      #   live_streaming_processing, and urgent_reference_processing.
+      def status
+        @status ||= @data["status"]
+      end
+
+      # @return [Boolean] whether the reference is activating.
+      def active?
+        status == 'activating'
+      end
```
Comment (claudiofullscreen) [inline lib/yt/models/reference.rb:83]: @stavro  Can I change this method to `activating?` ?

File: lib/yt/collections/references.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        super.tap do |params|
+          params[:params] = references_params
+          params[:path] = '/youtube/partner/v1/references'
+        end
+      end
+
+      def references_params
+        {onBehalfOfContentOwner: @parent.owner_name}.tap do |params|
+          (@extra_params ||= {}).each do |key, value|
+            params[key.to_s.camelize :lower] = value
```
Comment (claudiofullscreen) [inline lib/yt/collections/references.rb:31]: I was just working on making this sort of conversion universal, see https://github.com/Fullscreen/yt/commit/c33fc0368b093d9be9c8cfff704a281ee1334d26#diff-dd4964b049f643e9b1ac08ba94467d38R91

File: lib/yt/models/reference.rb (hunk 1 above)
Comment (stavro) [inline lib/yt/models/reference.rb:83]: Oops.. Yes please! Good catch.

File: lib/yt/models/reference.rb [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        @video_id ||= @data["videoId"]
+      end
+
+      # @return [String] the claim ID that represents the resulting association
+      #   between the asset and the video.
+      # @note This field is only present if the reference was created by associating
+      #   an asset with an existing YouTube video that was uploaded to a YouTube
+      #   channel linked to your CMS account.
+      def claim_id
+        @video_id ||= @data["videoId"]
```
Comment (claudiofullscreen) [inline lib/yt/models/reference.rb:43]: And I'll change this to `@claim_id`

File: lib/yt/models/reference.rb [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+      # @return [Boolean] the urgent status of the reference file
+      # @note Set this value to true to indicate that YouTube should prioritize 
+      #   Content ID processing for a video file. YouTube processes urgent video
+      #   files before other files that are not marked as urgent. This setting
+      #   is primarily used for videos of live events or other videos that require
+      #   time-sensitive processing. The sooner YouTube completes Content ID
+      #   processing for a video, the sooner YouTube can match user-uploaded videos 
+      #   to that video.
+      def urgent
```
Comment (claudiofullscreen) [inline lib/yt/models/reference.rb:69]: And this to `def urgent?`

File: lib/yt/models/reference.rb (hunk 3 above)
Comment (stavro) [inline lib/yt/models/reference.rb:43]: ehh I dropped the ball on that one. :sadpanda:

File: lib/yt/models/reference.rb [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+      #   channel linked to your CMS account.
+      def claim_id
+        @video_id ||= @data["videoId"]
+      end
+
+      # @return [boolean] whether or not the reference content is included in YouTube's
+      #   AudioSwap program.
+      # @note Set this field's value to true to indicate that the reference content
+      #   should be included in YouTube's AudioSwap program.
+      def audioswap_enabled
```
Comment (claudiofullscreen) [inline lib/yt/models/reference.rb:50]: And this to `audioswap_enabled?`

File: lib/yt/models/reference.rb [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        @audioswap_enabled ||= @data["audioswapEnabled"]
+      end
+
+      # @return [boolean] should the reference be used to generate claims.
+      # @note Set this value to true to indicate that the reference should not be used to 
+      #   generate claims. This field is only used on AudioSwap references.
+      def ignore_fp_match
```
Comment (claudiofullscreen) [inline lib/yt/models/reference.rb:57]: And this to `ignore_fp_match?`

**Comment by claudiofullscreen** [comment]
Replaced with https://github.com/Fullscreen/yt/pull/39

**Pull request closed without merging**